examples/custom_layout_manager/custom_layout_child.rs

```diff
@@ -0,0 +1,69 @@
+use gtk::{gdk, glib, graphene};
+use gtk::{prelude::*, subclass::prelude::ObjectSubclassExt};
+
+mod imp {
+    use super::*;
+    use gtk::subclass::prelude::*;
+    use std::cell::RefCell;
+
+    #[derive(Debug)]
+    pub struct CustomLayoutChild {
+        pub color: RefCell<gdk::RGBA>,
+    }
+    impl Default for CustomLayoutChild {
+        fn default() -> Self {
+            Self {
+                color: RefCell::new(gdk::RGBA::black()),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CustomLayoutChild {
+        const NAME: &'static str = "CustomLayoutChild";
+        type Type = super::CustomLayoutChild;
+        type ParentType = gtk::Widget;
+    }
+
+    impl ObjectImpl for CustomLayoutChild {
+        fn constructed(&self, widget: &Self::Type) {
+            widget.set_margin_top(4);
+            widget.set_margin_bottom(4);
+            widget.set_margin_start(4);
+            widget.set_margin_end(4);
+        }
+    }
+
+    impl WidgetImpl for CustomLayoutChild {
+        fn measure(
+            &self,
+            _widget: &Self::Type,
+            _orientation: gtk::Orientation,
+            _for_size: i32,
+        ) -> (i32, i32, i32, i32) {
+            // Return (minimum size, natural size, baseline position for the minimum size, baseline position for natural size)
+            (32, 32, -1, -1)
+        }
+
+        fn snapshot(&self, widget: &Self::Type, snapshot: &gtk::Snapshot) {
+            snapshot.append_color(
+                &self.color.borrow(),
+                &graphene::Rect::new(0.0, 0.0, widget.width() as f32, widget.height() as f32),
+            )
+        }
+    }
+}
+
+glib::wrapper! {
+    pub struct CustomLayoutChild(ObjectSubclass<imp::CustomLayoutChild>)
+        @extends gtk::Widget;
+}
+
+impl CustomLayoutChild {
+    pub fn new(color: gdk::RGBA) -> Self {
+        let widget = glib::Object::new(&[]).expect("Failed to create CustomLayoutChild");
+        let inner = imp::CustomLayoutChild::from_instance(&widget);
+        inner.color.replace(color);
```

Comment:
This should really be a construct-only property :)

Reply:
Does it has to be a property though? :)

Reply:
Updated as we don't have an example with a boxed property :-)

Reply:
> Does it has to be a property though? :)

Well, constructors should (ideally!) only call `glib::Object::new` and nothing else :)